datasets/greek_legal_code/greek_legal_code.py

```diff
@@ -0,0 +1,149 @@
+    year = "2021",
+    address = "Punta Cana, Dominican Republic",
+    publisher = "",
+    url = "",
+    doi = "",
+    pages = ""
+}
+"""
+
+_DESCRIPTION = """\
+Greek_Legal_Code contains 47k classified legal resources from Greek Legislation. Its origin is “Permanent Greek Legislation Code - Raptarchis”,
+a collection of Greek legislative documents classified into multi-level (from broader to more specialized) categories.
+"""
+
+_HOMEPAGE = "https://doi.org/10.5281/zenodo.5528002"
+
+_LICENSE = "CC BY-SA (Creative Commons / Attribution-ShareAlike)"
+
+URL = "https://zenodo.org/record/5528002/files/greek_legislation.tar.xz"
+
+
+class GreekLegalCodeConfig(datasets.BuilderConfig):
+    """BuilderConfig for Greek_Legal_Code."""
+
+    def __init__(self, label_type="volume", **kwargs):
+        """BuilderConfig for Greek_Legal_Code.
+
+        Args:
+        label_type: One of following: [volume, chapter, subject]
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(GreekLegalCodeConfig, self).__init__(**kwargs)
+        self.label_type = label_type
+
+
+class GreekLegalCode(datasets.GeneratorBasedBuilder):
+    """\
+    Greek_Legal_Code contains 47k classified legal resources from Greek Legislation. Its origin is “Permanent Greek Legislation Code - Raptarchis”,
+    a collection of Greek legislative documents classified into multi-level (from broader to more specialized) categories.
+    """
+
+    VERSION = datasets.Version("1.0.0")
+
+    BUILDER_CONFIGS = [
+        GreekLegalCodeConfig(
+            name=label_type,
+            label_type=label_type,
+            version=datasets.Version("1.0.0", ""),
+            description=f"Plain text import of Greek_Legal_Code for the {label_type} level",
+        )
+        for label_type in ["volume", "chapter", "subject"]
+    ]
+
+    DEFAULT_CONFIG_NAME = "volume"
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "text": datasets.Value("string"),
+                "label": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
```

Comment:
we can add a text classification template:
```suggestion
            citation=_CITATION,
            task_templates=[TextClassification(text_column="text", label_column="label")],
```